### PR TITLE
Include root cause reasons in Elasticsearch error message

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Include root cause reasons in Elasticsearch error message.

**Why?**
Some Elasticsearch errors are very generic and actionable error reason is in details.

**How did you test it?**
Existing tests.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: common/persistence/elasticsearch/visibility_store.go [hunk 1]
```diff
@@ -1197,3 +1197,24 @@ func parseHHMMSSDuration(d string) (int64, error) {
 
 	return hours*int64(time.Hour) + minutes*int64(time.Minute) + seconds*int64(time.Second) + nanos, nil
 }
+
+func detailedErrorMessage(err error) string {
```
Comment (yiminc) [inline common/persistence/elasticsearch/visibility_store.go:1201]: could you add a unit test for this method.

**Review by yiminc — APPROVED** [review]

File: common/persistence/elasticsearch/visibility_store.go (hunk 1 above)
Comment (alexshtin) [reply common/persistence/elasticsearch/visibility_store.go:1201]: Yes, I was about to do it. Adding.

File: common/persistence/elasticsearch/visibility_store.go [hunk 2]
```diff
@@ -1197,3 +1197,24 @@ func parseHHMMSSDuration(d string) (int64, error) {
...
+	elasticErr, isElasticErr := err.(*elastic.Error)
```
Comment (mmcshane) [inline common/persistence/elasticsearch/visibility_store.go:1202]: Can you make this more future-proof with `errors.As`?

Comment (wxing1292) [reply common/persistence/elasticsearch/visibility_store.go:1202]: > errors.As

what will errors.As provide? if not more complexity?

Comment (alexshtin) [reply common/persistence/elasticsearch/visibility_store.go]: It is canonical go way to check error type.

**Pull request merged**